DevProxy/Proxy/ProxyEngine.cs

```diff
@@ -615,21 +615,44 @@ private static void ToggleSystemProxy(ToggleSystemProxyAction toggle, string? ip
         process.WaitForExit();
     }
 
-    private static bool IsFirstRun()
+    internal static class HasRunFlag
     {
-        var firstRunFilePath = Path.Combine(ProxyUtils.AppFolder!, ".hasrun");
-        if (File.Exists(firstRunFilePath))
+        private static readonly string filename = Path.Combine(ProxyUtils.AppFolder!, ".hasrun");
+
+        public static bool IsFirstRun()

```

Comment:
Since the class is now named `HasRunFlag` let's rename this to `Exists` so that it's cleaner

Reply:
The Exists is confusing even more.  The func returns boolean depending on wheather the flag is created or not
So I am renaming IsFirstRun() to CreateFileIfMissing(). Let me know if it is not okay

Reply:
That's also nice. Good idea.